lib/auth/auth_with_roles_test.go

```diff
@@ -336,8 +337,54 @@ func TestListNodes(t *testing.T) {
 	require.Empty(t, cmp.Diff(expectedNodes, nodes))
 }
 
+// TestAPILockedOut tests Auth API when there are locks involved.
+func TestAPILockedOut(t *testing.T) {
```

Comment:
This test feels redundant since the Authorizer functionality is already tested by `TestAuthorizeWithLocksForLocalUser` in `permissions_test.go`. IIUC, the only difference is that this test checks whether `clt.GetUser` calls the authorizer.

Reply:
Yep, this is an additional test that `Authorize` gets invoked for actual client calls. Covering the case of `GetUser` is incidentally also useful because it involves [a pathway that could skip some checks](https://github.com/gravitational/teleport/blob/d96ea00a00c897ce2fed9f8dca92ca17932d8d02/lib/auth/auth_with_roles.go#L1317-L1324).